Sources/PrimerSDK/Classes/PCI/Services/RequestDispatcher.swift

```diff
@@ -0,0 +1,80 @@
+//
+//  RequestDispatcher.swift
+//  PrimerSDK
+//
+//  Created by Jack Newcombe on 14/03/2024.
+//
+
+import Foundation
+
+typealias DispatcherCompletion = (Result<DispatcherResponse, Error>) -> Void
+
+protocol RequestDispatcher {
+    func dispatch(request: URLRequest) async throws -> DispatcherResponse
+    func dispatch(request: URLRequest, completion: @escaping DispatcherCompletion) throws -> PrimerCancellable?
+}
+
+struct DispatcherResponseModel: DispatcherResponse {
+    let metadata: ResponseMetadata
+    let data: Data?
+    let error: Error?
+}
+
+struct ResponseMetadataModel: ResponseMetadata {
+    let responseUrl: String?
+    let statusCode: Int
+    let headers: [String: String]?
+}
+
+protocol DispatcherResponse {
+    var metadata: ResponseMetadata { get }
+    var data: Data? { get }
+    var error: Error? { get }
+}
+
+protocol URLSessionProtocol {
+    func dataTask(with request: URLRequest, completionHandler: @escaping @Sendable (Data?, URLResponse?, (any Error)?) -> Void) -> URLSessionDataTask
+}
+
+extension URLSession: URLSessionProtocol {}
+
+class DefaultRequestDispatcher: RequestDispatcher {
+
+    let urlSession: URLSessionProtocol
+
+    init(urlSession: URLSessionProtocol = URLSession.shared) {
+        self.urlSession = urlSession
+    }
+
+    func dispatch(request: URLRequest) async throws -> DispatcherResponse {
+        return try await withCheckedThrowingContinuation { continuation in
```

Comment:
Im wondering could we do this without the continuation? Is there something we want to maintain about the cancellable behaviour?

Example:
```swift
do {
    let (data, response) = try await urlSession.data(for: request)
    ...
    return .success(responseModel)
} catch {
    ...error handling
}
```